src/main/java/uk/gov/hmcts/divorce/common/event/Applicant2SolicitorApplyForFinalOrder.java

```diff
@@ -0,0 +1,239 @@
+        return new PageBuilder(configBuilder
+            .event(FINAL_ORDER_REQUESTED_APP2_SOL)
+            .forStates(AwaitingFinalOrder)
+            .name(APPLY_FOR_FINAL_ORDER)
+            .description(APPLY_FOR_FINAL_ORDER)
+            .showCondition("applicationType=\"soleApplication\" AND finalOrderReminderSentApplicant2=\"Yes\"")
+            .showSummary()
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE, APPLICANT_2_SOLICITOR)
+            .aboutToStartCallback(this::aboutToStart)
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grantHistoryOnly(
+                CASE_WORKER,
+                SUPER_USER,
+                LEGAL_ADVISOR,
+                APPLICANT_1_SOLICITOR,
+                JUDGE));
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
+        log.info("{} about to start callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        CaseData data = details.getData();
+        log.info("Retrieving order summary");
+        final OrderSummary orderSummary = paymentService.getOrderSummaryByServiceEvent(SERVICE_OTHER, EVENT_GENERAL, KEYWORD_NOTICE);
+
+        data.getFinalOrder().setApplicant2SolFinalOrderFeeOrderSummary(orderSummary);
+
+        data.getFinalOrder().setApplicant2SolFinalOrderFeeInPounds(
+            NumberFormat.getNumberInstance().format(
+                new BigDecimal(orderSummary.getPaymentTotal()).movePointLeft(2)
+            )
+        );
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(data)
+            .errors(null)
+            .warnings(null)
+            .build();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} about to submit callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        CaseData data = details.getData();
+        data.getApplication().setPreviousState(beforeDetails.getState());
+
+        final List<String> errors = applyForFinalOrderService.validateApplyForFinalOrder(data, true);
+        if (!errors.isEmpty()) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(data)
+                .errors(errors)
+                .build();
+        }
+
+        CaseDetails<CaseData, State> updatedDetails = applyForFinalOrderService.applyForFinalOrderAsApplicant2(details);
+        CaseData updatedData = updatedDetails.getData();
+        var updatedFo = updatedData.getFinalOrder();
+
+        final var finalOrderFeeOrderSummary = updatedFo.getApplicant2SolFinalOrderFeeOrderSummary();
+
+        if (updatedFo.isSolicitorPaymentMethodPba()) {
```

Comment:
Could this PBA section be put into a private method below the aboutToSubmit? Would make it cleaner to follow.